administrator/components/com_guidedtours/src/Model/StepModel.php

```diff
@@ -0,0 +1,391 @@
+     * @return  void
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function populateState()
+    {
+        parent::populateState();
+
+        $app       = Factory::getApplication();
+        $context   = $this->option . '.' . $this->name;
+    }
+
+    protected function generateNewTitle($categoryId, $alias, $title)
+    {
+        // Alter the title
+        $table = $this->getTable();
+
+        while ($table->load(array('title' => $title))) {
+            $title = StringHelper::increment($title);
+        }
+
+        return array($title, $alias);
+    }
+
+    /**
+     * Method to save the form data.
+     *
+     * @param   array  $data  The form data.
+     *
+     * @return   boolean  True on success.
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function save($data)
+    {
+        $table      = $this->getTable();
+        $context    = $this->option . '.' . $this->name;
+        $app        = Factory::getApplication();
+        $user       = $app->getIdentity();
+        $input      = $app->input;
+        $tourID = $app->getUserStateFromRequest($context . '.filter.tour_id', 'tour_id', 0, 'int');
+
+        if (empty($data['tour_id'])) {
+            $data['tour_id'] = $tourID;
+        }
+
+        $tour = $this->getTable('Tour');
+
+        $tour->load($data['tour_id']);
+
+        $parts = explode('.', $tour->extension);
+
+        if (isset($data['rules']) && !$user->authorise('core.admin', $parts[0])) {
+            unset($data['rules']);
+        }
+
+        // Make sure we use the correct extension when editing an existing tour
+        $key = $table->getKeyName();
+        $pk  = (isset($data[$key])) ? $data[$key] : (int) $this->getState($this->getName() . '.id');
+
+        if ($pk > 0) {
+            $table->load($pk);
+
+            if ((int) $table->tour_id) {
+                $data['tour_id'] = (int) $table->tour_id;
+            }
+        }
+
+        if ($input->get('task') == 'save2copy') {
+            $origTable = clone $this->getTable();
+
+            // Alter the title for save as copy
+            if ($origTable->load(['title' => $data['title']])) {
+                list($title) = $this->generateNewTitle(0, '', $data['title']);
+                $data['title'] = $title;
+            }
+
+            $data['published'] = 0;
+            $data['default']   = 0;
+        }
+
+        return parent::save($data);
+    }
+
+    /**
+     * Method to change the default state of one item.
+     *
+     * @param   array    $pk     A list of the primary keys to change.
+     * @param   integer  $value  The value of the home state.
+     *
+     * @return  boolean  True on success.
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function setDefault($pk, $value = 1)
+    {
+        $table = $this->getTable();
+
+        if ($table->load($pk)) {
+            if ($table->published !== 1) {
+                $this->setError(Text::_('COM_WORKFLOW_ITEM_MUST_PUBLISHED'));
```

Comment:
wrong error message - this component doesnt use workflows

Reply:
Removed the function